core/SupportedBrowser.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik;
+
+use Piwik\Http;
+use Piwik\Piwik;
+use Piwik\Container\StaticContainer;
+use Piwik\DeviceDetector\DeviceDetectorFactory;
+use Piwik\Exception\NotSupportedBrowserException;
+
+class SupportedBrowser
+{
+    /**
+     * A list of browsers with version numbers that are not supported. A browser
+     * is not supported if it's included in this array and the version number
+     * is smaller or equal to the number in this array. If the user's browser
+     * not included in this list or the version number is higher, then
+     * supported.
+     *
+     * Current version numbers are coming from this list:
+     * https://caniuse.com/rel-noreferrer
+     */
+    private static $notSupportedBrowsers = [
+        'FF' => 32,
+        'IE' => 10,
+        'SF' => 4,
+        'CH' => 15,
+        'OP' => 12,
+        'PS' => 12,
+    ];
+
+    public static function checkIfBrowserSupported()
+    {
+        $userAgent = Http::getUserAgent();
```

Comment:
btw @flamisz  it may not be the best method to get the user agent actually as it's only meant for the HTTP class as it returns a default user agent if none is set. If I see this right there is no method for this actually yet so we could just use something like `
        $userAgent = isset($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : '';`  and then if the user agent is empty then we'd directly `return true`. Or we could check in `isSupportedBrowserCheckNeeded` method if a user agent is set and return true/false there?

Reply:
yes, you are right. `$_SERVER['HTTP_USER_AGENT']` is the good solution here.